### PR TITLE
Extract publishing and animalsniffer plugins.

### PR DESCRIPTION
**Pull request by anuraaga**
Also fixed a couple of deprecations

### CODE REVIEW

File: buildSrc/src/main/kotlin/otel.publish-conventions.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+if (System.getenv("CI") != null) {
+    signing {
+        useInMemoryPgpKeys(System.getenv("GPG_PRIVATE_KEY"), System.getenv("GPG_PASSWORD"))
+        sign(publishing.publications["mavenPublication"])
+    }
+}
+
+tasks {
+    register("updateVersionInDocs") {
```
Comment (anuraaga) [inline buildSrc/src/main/kotlin/otel.publish-conventions.gradle.kts:82]: We previously added this to allprojects but it seems to only be needed for published ones

File: buildSrc/src/main/kotlin/otel.japicmp-conventions.gradle.kts [hunk 2]
```diff
@@ -42,35 +42,38 @@ fun findArtifact(version: String): File {
...
-    tasks {
-        val jApiCmp by registering(JapicmpTask::class) {
-            dependsOn("jar")
-            // the japicmp "old" version is either the user-specified one, or the latest release.
-            val apiBaseVersion: String? by project
-            val baselineVersion = apiBaseVersion ?: latestReleasedVersion
-            val baselineArtifact = findArtifact(baselineVersion)
-            oldClasspath = files(baselineArtifact)
+// generate the api diff report for any module that is stable and publishes a jar.
+if (!project.hasProperty("otel.release") && !project.name.startsWith("bom")) {
```
Comment (anuraaga) [inline buildSrc/src/main/kotlin/otel.japicmp-conventions.gradle.kts:46]: This seems like a "feature" of the japicmp conventions, not publishing ones so didn't use the conditional plugin application like before

**Review by kubawach — APPROVED** [review]
Nice :)

**Review by jkwatson — APPROVED** [review]
Thanks!

**Pull request merged**